minerva/minerva.py

```diff
@@ -158,6 +159,28 @@ async def on_message(self, update: Update, context: ContextTypes.DEFAULT_TYPE) -
       # If the user uploads multiple photos in a single message, Telegram API
       # will emit different message events for them
       # TODO(yurij): wait for other photos here
+    elif message.document and message.document.mime_type == "text/plain":
+      file = await message.document.get_file()
+      file_content = (await file.download_as_bytearray()).decode("utf-8", errors="ignore")
+      topic_id = self._get_topic_id(message)
+      
+      # Context to only awknowledge the upload and wait for followup 
+      # as telegram does not allow to tag bot in document caption
+      history_message = Message(
+          author=f"file-{message.from_user.username or message.from_user.id}",
```

Comment:
Let's make sure to use the `message_author` as author. It will help it tie messages sent by the user to the file sent by the same user.

```suggestion
          author=message_author,
```